analyses/tp53_nf1_score/00-tp53-nf1-alterations.R

```diff
@@ -0,0 +1,103 @@
+# to evaluate classifier
+# @params snvConcensus multi-caller concensus snv calls
+# @params clincalFile clinical file: pbta-histologies.tsv
+# @params outputFolder output folder for alteration file
+# @params gencode cds bed file from gencode
+
+suppressPackageStartupMessages(library("optparse"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("GenomicRanges"))
+
+#### Source functions ----------------------------------------------------------
+# We can use functions from the `snv-callers` module of the OpenPBTA project
+# TODO: if a common util folder is established, use that instead
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+source(file.path(root_dir, "analyses", "snv-callers", "util",
+                 "tmb_functions.R"))
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-s", "--snvConsensus"),type="character",
+              help="Consensus snv calls (.tsv) "),
+  make_option(c("-c","--clinicalFile"),type="character",
+              help="clinical file for all samples (.tsv)"),
+  make_option(c("-o","--outputFolder"),type="character",
+              help="output folder for results "),
+  make_option(c("-g","--gencode"),type="character",
+              help="cds gencode bed file")
+)
+
+opt <- parse_args(OptionParser(option_list=option_list))
+snvConsensusFile <- opt$snvConsensus
+clinicalFile <- opt$clinicalFile
+outputFolder <- opt$outputFolder
+gencodeBed <- opt$gencode
+
+#### Generate files with TP53, NF1 mutations -----------------------------------
+
+# read in consensus SNV files
+consensus_snv <- read_tsv(snvConsensusFile)
+# gencode cds region BED file
+gencode_cds <- read_tsv(gencodeBed, col_names = FALSE)
+# clinical file
+clinical <- read_tsv(clinicalFile)
+
+# filter the MAF data.frame to only include entries that fall within the
+# CDS bed file regions
+coding_consensus_snv <- snv_ranges_filter(maf_df = consensus_snv,
+                                          keep_ranges = gencode_cds)
+
+# subset to TP53, removing silent mutations and mutations in introns
+tp53_coding <- coding_consensus_snv %>%
+  filter(Hugo_Symbol == "TP53") %>%
+  filter(!(Variant_Classification %in% c("Silent", "Intron")))
+
+# subset to NF1, removing silent mutations, mutations in introns, and missense
+# mutations -- we exclude missense mutations because they are not annotated
+# with OncoKB
+# https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/381#issuecomment-570748578
+nf1_coding <- coding_consensus_snv %>%
+  filter(Hugo_Symbol == "NF1") %>%
+  filter(!(Variant_Classification %in% c("Silent",
+                                         "Intron",
+                                         "Missense_Mutation")))
+
+# include only the relevant columns from the MAF file
+tp53_nf1_coding <- tp53_coding %>%
```

Comment:
If these two data.frames are going to bound together anyway, is there any reason you can't just do one filtering step where you select both NF1 and TP53? 
```suggestion
tp53_nf1_coding <- coding_consensus_snv %>%
  filter(Hugo_Symbol %in% c("NF1", "TP53")) %>%
  filter(!(Variant_Classification %in% c("Silent",
                                         "Intron",
                                         "Missense_Mutation")))
```

If the concern is getting Missense_Mutations that are TP53 then you could add an another filter step to remove that combo. e.g. `dplyr::filter(!(Hugo_Symbol == "TP53" && Variant_Classification == "Missense_Mutation")`
If you are going to use this logic piece^ I would double check that it gives you what you want.

Reply:
I added this as two, more explicit steps on purpose because it's important to document the logic around _NF1_ specifically and because multiple people are working on this module.